### PR TITLE
feat: implements text/event-stream(SSE) MIME parser

### PR DESCRIPTION
**Pull request by jizhuozhi**
### Ⅰ. Describe what this PR did

Implements an EventStream for common SSE(text/event-stream) parsing.

And add sse-timing plugin for tracing timestamp that each event received in higress as an example. As an alternative implementation of the server-timing protocol (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing). It's useful to profile AI backend latency which uses event-stream MIME type. 

### Ⅱ. Does this pull request fix one issue?

None

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
As an example, there is no need to integration test, but has an unit test in `event_stream.rs`

### Ⅳ. Describe how to verify it

There is a docker-compose.yml in plugin directory, with a go server backend which provides sse response, up it and then just use broswer to open http://localhost:10000, you could see events in dev-tools.

### CODE REVIEW

**Comment by jizhuozhi** [comment]
An alternative PR of https://github.com/alibaba/higress/pull/1361, but not as a publish plugin

**Comment by johnlanni** [comment]
cc @007gzs

**Comment by jizhuozhi** [comment]
@johnlanni @007gzs PTAL

File: plugins/wasm-rust/src/event_stream.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        } else {
+            None
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
```
[the remark this answers is not in the document]
Comment (007gzs) [reply plugins/wasm-rust/src/event_stream.rs:117]: @jizhuozhi

**Review by 007gzs — CHANGES_REQUESTED** [review]

File: plugins/wasm-rust/src/event_stream.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jizhuozhi) [reply plugins/wasm-rust/src/event_stream.rs:117]: Will, according to ABNF of `text/event-stream`, `event6\r\n` is a valid event which `\r\n` has been consumed in `event6\r\n`. Follow the principle of issuing valid data as early as possible, `\r\n` should not belong to `event6\r\n` but as an independent empty event. So this test case should be 
```go
    fn test_mixed2_eol_events() {
        let mut parser = EventStream::new();
        parser.update(b"event5\r\nevent6\r\n".to_vec());
        assert_eq!(parser.next(), Some(b"event5".to_vec()));
        assert_eq!(parser.next(), Some(b"event6".to_vec()));
        parser.update(b"\r\nevent7\r\n".to_vec());
        assert_eq!(parser.next(), Some(b"".to_vec()));
        assert_eq!(parser.next(), Some(b"event7".to_vec()));
    }
```

[the remark this answers is not in the document]
Comment (jizhuozhi) [reply plugins/wasm-rust/src/event_stream.rs:117]: This is a constraint: all valid events have been consumed after `next`, and empty event is valid (`event = *( comment / field ) eol`)

[the remark this answers is not in the document]
Comment (007gzs) [reply plugins/wasm-rust/src/event_stream.rs:117]: I think the results of `test_mixed2_eol_events` and `test_mixed_eol_events`should have the same return
```
    #[test]
    fn test_mixed_eol_events() {
        let mut parser = EventStream::new();
        parser.update(b"event5\r\nevent6\r\n\r\nevent7\r\n".to_vec());

        assert_eq!(parser.next(), Some(b"event5".to_vec()));
        assert_eq!(parser.next(), Some(b"event6".to_vec()));
        assert_eq!(parser.next(), Some(b"event7".to_vec()));
    }
```
https://github.com/alibaba/higress/pull/1416/files#diff-0062d0a4075996e96d98a35bdc644756c25726e8b7f110037570a36daece0642R147-R155

[the remark this answers is not in the document]
Comment (jizhuozhi) [reply plugins/wasm-rust/src/event_stream.rs:117]: Attention: `next` has side-effect and CRLF in ABNF is an ambiguous grammar, and empty event is accepted by`event = *( comment / field ) eol` ,we have already addressed 
```
In order to eliminate this ambiguity, we believe that CRLF as CR+LF belongs to event and field respectively
```
What we can do is eliminate this ambiguity, but it's users' responsibility to judge how to process empty event.

**Review by 007gzs — APPROVED** [review]
LGTM

**Pull request merged**